NuRadioReco/modules/voltageToAnalyticEfieldConverterNeutrinos.py

```diff
@@ -0,0 +1,546 @@
+			else:
+				E_theta = params[0]
+				theta = params[1]
+
+			if phasing: E_theta *= len(phasing_dict[channel_Vpol])*np.sqrt(2)
+			time_trace_thet = get_time_trace(E_theta, theta, N_V_time_domain, 
+											 dt, shower_type, n_index, R,  parametrization)
+			trace_freq_thet = fft.time2freq(time_trace_thet, 1/dt)
+			trace_freq_phi = np.zeros(len(trace_freq_thet))
+
+			if attenuation: 
+				trace_freq_thet *= attn
+			if bandpass: 
+				trace_freq_thet *= h
+
+			chi2 = 0
+			for iCh, trace in enumerate(V_timedomain):
+				if use_channels_efield[iCh] == channel_Vpol:
+					if phasing: trace = get_phased_trace(use_channels_efield[iCh], 
+														 phasing_dict[use_channels_efield[iCh]])
+					if bandpass:
+						f = fft.time2freq(trace, 1/dt) * h
+						trace = fft.freq2time(f, 1/dt)
+					k = np.argmax(trace)
+					analytic_trace_fft = np.sum(efield_antenna_factor[iCh] * np.array([trace_freq_thet, trace_freq_phi]), axis = 0)
+					analytic_trace = fft.freq2time(analytic_trace_fft, 1/dt)
+
+					if hilbert: 
+						chi2 += np.sum(abs((abs(signal.hilbert(np.roll(analytic_trace, 
+																	   int(toffset))))[k - N: k+2*N] - abs(signal.hilbert(trace))[k - N: k+2*N]))**2)
+					else:
+						chi2 += np.sum((np.roll(analytic_trace, 
+												int(toffset))[k-N:k+3*N] - trace[k-N:k+3*N])**2)
+					
+			return chi2/(2*noise_RMS**2)
+
+		def minimizer(params, R, n_index, toffset, 
+					  hilbert = False, minimizer = True):
+			
+			"""
+			params: Array of floats
+				parameters to be minimized 
+			R: float
+				trajectory length 
+			n_index: float
+				refractive index 
+			toffset: float
+				time shift for the analytic trace 
+			hilbert: bool
+				if True, the hilbert envelop is used for the fit 
+			minimizer: bool
+				if True, the output of minimization is returned. 
+				if False, the sign of the Hpol component is returned. 
+			"""
+			N = 50
+		
+			if len(params) == 3:
+				E_theta, theta, E_phi = params
+			else:
+				E_phi = params[0]
+				E_theta = results.x[0]
+				theta = results.x[1]
+
+			if phasing: E_phi *= len(phasing_dict[channel_Hpol])*np.sqrt(2)
+			chi2 = np.inf
+
+			for sign_tmp in [-1, 1]:
+				tmp = 0
+				time_trace_thet = get_time_trace(E_theta, theta, N_V_time_domain, 
+												 dt, shower_type, n_index, R, parametrization)
+				trace_freq_thet = fft.time2freq(time_trace_thet, 1/dt)
+				time_trace_phi = sign_tmp*get_time_trace(E_phi, theta, N_V_time_domain, 
+														 dt, shower_type, n_index, R, parametrization)
+				trace_freq_phi = fft.time2freq(time_trace_phi, 1/dt)
+
+				if attenuation: 
+					trace_freq_thet *= attn
+					trace_freq_phi *= attn
+				if bandpass: 
+					trace_freq_thet *= h
+					trace_freq_phi *= h
+				
+				for iCh, trace in enumerate(V_timedomain):
+					if use_channels_efield[iCh] == channel_Vpol:
+						k = np.argmax(trace)
+					if use_channels_efield[iCh] == channel_Hpol:
+						if phasing: trace = get_phased_trace(use_channels_efield[iCh],
```

Comment:
I think this variable isn't actually used later

Reply:
What do you mean here?